homeassistant/components/binary_sensor/esphome.py

```diff
@@ -0,0 +1,63 @@
+"""Support for ESPHome binary sensors."""
+import logging
+
+from typing import TYPE_CHECKING, Optional
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.esphome import EsphomeEntity, \
+    platform_async_setup_entry
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from aioesphomeapi import BinarySensorInfo, BinarySensorState  # noqa
+
+DEPENDENCIES = ['esphome']
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up ESPHome binary sensors based on a config entry."""
+    # pylint: disable=redefined-outer-name
+    from aioesphomeapi import BinarySensorInfo, BinarySensorState  # noqa
+
+    await platform_async_setup_entry(
+        hass, entry, async_add_entities,
+        component_key='binary_sensor',
+        info_type=BinarySensorInfo, entity_type=EsphomeBinarySensor,
+        state_type=BinarySensorState
+    )
+
+
+class EsphomeBinarySensor(EsphomeEntity, BinarySensorDevice):
+    """A binary sensor implementation for ESPHome."""
+
+    @property
+    def _static_info(self) -> 'BinarySensorInfo':
+        return super()._static_info
+
+    @property
+    def _state(self) -> Optional['BinarySensorState']:
+        return super()._state
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        if self._static_info.is_status_binary_sensor:
```

Comment:
Is there really a need for an availability binary sensor if entities just become unavailable?

Reply:
Then you can get nice graphs like this:

<img width="364" alt="screen shot 2018-12-17 at 15 36 57" src="https://user-images.githubusercontent.com/6833237/50093673-a3f5b800-0211-11e9-88a7-0e76f84f431a.png">

with `sensor` you pretty much can't see the small blip when it's unavailable. Another option would be to declare a `binary_sensor` that's always on, but 
  - a) that's quite a hack and 
  - b) these status binary sensors have device class `connectivity`. For device class connectivity `ON` means connected and `OFF` means disconnected. unavailable doesn't really have a meaning for those.